python/friesian/example/multi_task/data_processing.py

```diff
@@ -0,0 +1,159 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import argparse
+import os
+from argparse import ArgumentParser
+
+from bigdl.friesian.feature import FeatureTable
+from bigdl.orca import init_orca_context, stop_orca_context
+
+
+def transform(x):
+    if x == '上海':
+        return 0.0
+    elif isinstance(x, float):
+        return float(x)
+    else:
+        return float(eval(x))
+
+
+def transform_cat_2(x):
+    return '-'.join(sorted(x.split('/')))
+
+
+def read_and_split(data_input_path, sparse_int_features, sparse_string_features, dense_features):
+    header_names = ['user_id', 'article_id', 'expo_time', 'net_status', 'flush_nums',
+                    'exop_position', 'click', 'duration', 'device', 'os', 'province', 'city',
+                    'age', 'gender', 'ctime', 'img_num', 'cat_1', 'cat_2'
+                    ]
+    if data_input_path.split('.')[-1] == 'csv':
+        data_pd = FeatureTable.read_csv(data_input_path, header=False, names=header_names)
+    else:
+        data_pd = FeatureTable.read_parquet(data_input_path)
+    data_pd = data_pd.cast(sparse_int_features, 'string')
+    data_pd = data_pd.cast(dense_features, 'string')
```

Comment:
why do you cast these features to string?

Reply:
For sparse_int_features, I want to call encode_string next to handle all sparse features; 
For dense_features, the data format is not uniform in the dataset, so the conversion process is: raw -> string -> float

Reply:
For the ids, reasonable to treat them as string since the id is very large, just an id, not index